components/admin_console/system_users/system_users.jsx

```diff
@@ -32,6 +32,26 @@ export default class SystemUsers extends React.Component {
          */
         teams: PropTypes.arrayOf(PropTypes.object).isRequired,
 
+        /**
+         * Title of the app or site.
+         */
+        siteName: PropTypes.string,
+
+        /**
+         * Whether or not MFA is licensed and enabled.
+         */
+        mfaEnabled: PropTypes.bool.isRequired,
```

Comment:
Just a little worried about `mfaEnabled` as a combination of licensed and enabled. Would it make sense to have separate props for `isLicensed` so that the consuming component will independently handle the logic rather than the higher order component?

Reply:
That's a good general question that I wrestled with in making these changes. On the one hand, I've tried to preserve all the original "bits" to keep the review simpler, but in reality, our selectors are really handling a lot of this kind of business logic already. In this case, having to pass down three bits multiple layers just to compute a single value seemed a waste, so I experimented with an "optimization" as it were.

In general, I'd be in favour of simplifying the surface area of our component interfaces. For example, I don't think it's really necessary for components to have to check the license and the configuration bit to decide if MFA is enabled /unless/ they handle the potential additional states themselves. I didn't do this holistically, of course. Thoughts?

Reply:
You have a good point there. Thanks!